src/posix/platform/trel.cpp

```diff
@@ -507,6 +519,12 @@ void platformTrelInit(const char *aTrelUrl)
 
     InitPacketQueue();
     sInitialized = true;
+
+    sCounters.mTxPackets = 0;
+    sCounters.mTxBytes   = 0;
+    sCounters.mRxPackets = 0;
+    sCounters.mRxBytes   = 0;
+    sCounters.mTxFailure = 0;
```

Comment:
Consider using `Clearable`. 

https://github.com/openthread/openthread/blob/1e82c4e9620e6ab708e94263641674052af9ec30/src/core/net/dnssd_server.hpp#L80C3-L80C3

Reply:
It seems we have a standard that counters share the lifespan with an OT instance, so it should outlive the trel enable and disable cycle?

Reply:
If this is in the non-platform code like trel_interface.hpp, we can define Counter to define Clearable. But platform/trel.h can be potentially a c header when compiling with simulation/trel.c? which doesn't understand ot namespace and class.

Regarding whether it should be cleared whenever trel is disabled. It seems that we have a tradition that counters share the same life span with OT instance, so we shouldn't clear the counters when disabling trel?

Reply:
Done.

Reply:
> If this is in the non-platform code like trel_interface.hpp, we can define Counter to define Clearable. But platform/trel.h can be potentially a c header when compiling with simulation/trel.c? which doesn't understand ot namespace and class.

The `Clearable` is used for simplifying your implementation `src/posix/platform/trel.cpp`. You don't have to modify your API struct `otPlatTrelCounters` to support `Clear()`.  Instead, you can define a class in `src/posix/platform/trel.cpp` which extends both `otPlatTrelCounters` and `Clearable`.  

See https://github.com/openthread/openthread/blob/1e82c4e9620e6ab708e94263641674052af9ec30/src/core/net/dnssd_server.hpp#L80C3-L80C3. Let me know if you have any further concerns.

Reply:
Thanks, done.

Reply:
If we have a type in OT core, then I would agree with @superwhd that we should use `Clearable` for it (if we want to set all member fields to zero).

But suggest not using `ot::Clearable` in the platform layer code (e.g., `posix`) since `common/clearable.h` is not a public OT header/definition to avoid adding dependency.

Reply:
> When trel is disabled, will the stale counters still be uploaded? What are they used for?

Would you please please help to clarify? @yangliu1333

Reply:
So as Abtin's suggested, since we support trel counters reset, users can choose to reset the counters or not. We can keep the counters when disabling the trel.

Reply:
```
Do we want to reset the counters when TREL is disabled? Or keep it going?
Or can add a new API to allow user reset it?

Also suggest we document the expected behavior in platform/trel.h so a platform developer knows what they are expected to write?
```

The related comment from abtin I can find is as above, I am fine to keep the counters in OT when trel is disabled, however my concern is whether or not the stale counters will be uploaded, note that metrics uploading can be thought as platform efforts.